samples/nrf9160/modem_shell/src/uart/uart_shell.c

```diff
@@ -69,7 +73,64 @@ void uart_toggle_power_state_at_event(const struct shell *shell, const struct lt
 		disable_uarts();
 	} else if (evt->type == LTE_LC_EVT_MODEM_SLEEP_EXIT) {
 		enable_uarts();
-		shell_print(shell, "Modem sleep exit: re-enabling UARTs");
+	}
+}
+
+void uart_toggle_power_state(const struct shell *shell)
+{
+	const struct device *uart_dev;
+	uint32_t uart0_power_state;
+	int err;
+
+	uart_dev = device_get_binding(DT_LABEL(DT_NODELABEL(uart0)));
+	if (uart_dev) {
+		err = pm_device_state_get(uart_dev, &uart0_power_state);
+		if (err) {
+			shell_print(shell,
+				   "Failed to change UART power state, device_get_power_state: %d.",
+				    err);
+			return;
+		}
+
+		if (uart0_power_state == PM_DEVICE_STATE_ACTIVE) {
+			shell_print(shell, "Disabling UARTs.");
+
+			/* allow little time for printing the notification */
+			k_sleep(K_MSEC(500));
+
+			/* set uart0 to low power state */
+			err = pm_device_state_set(uart_dev, PM_DEVICE_STATE_LOW_POWER, NULL, NULL);
+
+			/* set uart1 to low power state */
+			uart_dev = device_get_binding(DT_LABEL(DT_NODELABEL(uart1)));
+			if (uart_dev) {
+				err |= pm_device_state_set(uart_dev,
+							   PM_DEVICE_STATE_LOW_POWER,
+							   NULL,
+							   NULL);
+			}
+		} else {
+			/* set uart0 to active state */
+			err = pm_device_state_set(uart_dev, PM_DEVICE_STATE_ACTIVE, NULL, NULL);
+
+			/* set uart1 to active state */
+			uart_dev = device_get_binding(DT_LABEL(DT_NODELABEL(uart1)));
+			if (uart_dev) {
+				err |= pm_device_state_set(uart_dev,
+							   PM_DEVICE_STATE_ACTIVE,
+							   NULL,
+							   NULL);
+			}
+			shell_print(shell, "Enabling UARTs.");
+
+			/* stop timer if uarts were disabled with command 'uart disable' */
+			k_timer_stop(&uart_reenable_timer);
+		}
+		if (err) {
+			shell_print(shell, "Failed to change UART power state.");
+		}
```

Comment:
If the board has only one UART enabled, this would most likely print an error. I didn't have error prints in the original implementation because of this, just tried to disable both UARTs and that's it. Just wondering if it would be better to just remove checks for the `pm_device_state_set()` calls.

Reply:
Agreed. What if keep the error print on line 130, but update the err variable only when pm_device_state_set() is called for uart0?

Reply:
Sounds good.